sfge/src/main/java/com/salesforce/cli/FilesToEntriesMap.java

```diff
@@ -0,0 +1,120 @@
+package com.salesforce.cli;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+/**
+ * Mapping of files that occur in an {@link com.salesforce.graph.ApexPath} and the entry method from
+ * which the path originates.
+ */
+public class FilesToEntriesMap {
+    @VisibleForTesting static final String FIELD_DATA = "data";
+    @VisibleForTesting static final String FIELD_FILENAME = "filename";
+    @VisibleForTesting static final String FIELD_ENTRIES = "entries";
+    private final HashMap<String, HashSet<Entry>> map;
+
+    public FilesToEntriesMap() {
+        this.map = new HashMap<>();
+    }
+
+    /**
+     * Creates a new mapping of the file name and the corresponding entry method it impacts
+     *
+     * @param filename that a path touches
+     * @param entryFile of the entry method
+     * @param entryMethod of the entry method
+     */
+    public void put(String filename, String entryFile, String entryMethod) {
```

Comment:
The cache file only contains these three information right? No information on the violations?

Reply:
Right, no violation information at this time.